### PR TITLE
feat(python): Support explicit Binary dtype in constructor

### LINKED ISSUE #7303: Cannot initialize DataFrame with `Binary` data type in `schema` arg
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

I cannot pass `Binary` as a dtype in the `schema` argument of the `DataFrame` constructor. If I don't pass the schema argument, it auto-infers the dtype just fine.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({"a": [b"hello", b"world"]}, schema={"a": pl.Binary})
# ValueError: Cannot construct PySeries for type Binary.

s = pl.Series([b"hello", b"world"], dtype=pl.Binary)
# ValueError: Cannot construct PySeries for type Binary.

# These work fine
df = pl.DataFrame({"a": [b"hello", b"world"]})
s = pl.Series([b"hello", b"world"])
```

### Expected behavior

The code should run normally.

### Installed versions

master branch

### PR DESCRIPTION
**Pull request by ghuls**
Closes #7303

### CODE REVIEW

**Comment by ghuls** [comment]
This closes: https://github.com/pola-rs/polars/issues/7303

File: py-polars/tests/unit/test_constructors.py [hunk 1]
```diff
@@ -923,3 +923,13 @@ def test_array_to_pyseries_with_one_chunk_does_not_copy_data() -> None:
...
+
+
+def test_init_with_explicit_binary_schema() -> None:
+    df = pl.DataFrame({"a": [b"hello", b"world"]}, schema={"a": pl.Binary})
+    assert df.schema == {"a": pl.Binary}
+    assert df.to_arrow()["a"].to_pylist() == [b"hello", b"world"]
+
+    s = pl.Series("a", [b"hello", b"world"], dtype=pl.Binary)
+    assert s.dtype == pl.Binary
+    assert s.to_arrow().to_pylist() == [b"hello", b"world"]
```
Comment (stinodego) [inline py-polars/tests/unit/test_constructors.py:935]: Any reason why you're going through Arrow to test this? `s.to_list()` would also work, right?

```suggestion
    assert s.to_list() == [b"hello", b"world"]
```

File: py-polars/tests/unit/test_constructors.py [hunk 2]
```diff
@@ -923,3 +923,13 @@ def test_array_to_pyseries_with_one_chunk_does_not_copy_data() -> None:
         pyseries.get_chunks()[0]._get_ptr()
         == original_array.chunks[0].buffers()[1].address
     )
+
+
+def test_init_with_explicit_binary_schema() -> None:
+    df = pl.DataFrame({"a": [b"hello", b"world"]}, schema={"a": pl.Binary})
+    assert df.schema == {"a": pl.Binary}
+    assert df.to_arrow()["a"].to_pylist() == [b"hello", b"world"]
```
Comment (stinodego) [inline py-polars/tests/unit/test_constructors.py:931]: Same thing here.

```suggestion
    assert df["a"].to_list() == [b"hello", b"world"]
```

**Review by stinodego — COMMENTED** [review]
Nice, simple fix! Just one stylistic comment about the tests.

> This closes: https://github.com/pola-rs/polars/issues/7303

If you add this to the PR description (without the colon) it auto-links the issue, which is nice. I edited the description for now.

File: py-polars/tests/unit/test_constructors.py (hunk 1 above)
Comment (ghuls) [reply py-polars/tests/unit/test_constructors.py]: It was a brain fart. As Polars currently does not display a binary column/Series in a meaningful way, I didn't think about using `to_list()`. It is fixed now.

File: py-polars/tests/unit/test_constructors.py (hunk 2 above)
Comment (ghuls) [reply py-polars/tests/unit/test_constructors.py]: done.

**Review by stinodego — APPROVED** [review]
🎉

**Pull request merged**